core/embed/rust/src/ui/flow/swipe.rs

```diff
@@ -234,41 +219,20 @@ impl SwipeFlow {
 
                     return_transition = AttachType::Swipe(dir);
 
-                    let states_num = self.internal_pages;
-                    if states_num > 0 {
-                        if config.has_horizontal_pages() {
-                            let current_state = self.internal_state;
-                            if dir == SwipeDirection::Left && current_state < states_num - 1 {
-                                self.internal_state += 1;
-                                state_change = self.state_unchanged();
-                                attach = true;
-                            } else if dir == SwipeDirection::Right && current_state > 0 {
-                                self.internal_state -= 1;
-                                state_change = self.state_unchanged();
-                                attach = true;
-                            }
-                        }
-                        if config.has_vertical_pages() {
-                            let current_state = self.internal_state;
-                            if dir == SwipeDirection::Up && current_state < states_num - 1 {
-                                self.internal_state += 1;
-                                state_change = self.state_unchanged();
-                                attach = true;
-                            } else if dir == SwipeDirection::Down && current_state > 0 {
-                                self.internal_state -= 1;
-                                state_change = self.state_unchanged();
-                                attach = true;
-                            }
-                        }
+                    let new_internal_state =
+                        config.paging_event(dir, self.internal_state, self.internal_pages);
```

Comment:
While at it, can we please rename `internal_state` to something else? From the current name it's not clear it's related to `internal_pages`, and also "state" everywhere else is usually an enum and not integer. Perhaps:
* `internal_state` -> `current_internal_page`
* `internal_pages` -> `internal_page_count`

Reply:
i'd prefer to leave this alone here

(a) i have a draft of complete removal of internal state here
(b) after your suggestion, I actually tried the following:

```
#[derive(Copy, Clone)]
pub struct Pager {
   current: u16,
   total: u16,
}
```
which I'd like to use pervasively for paging, but that is definitely its own separate PR

Reply:
Agreed with the `Pager`, or something similar. It would be nice to unify the way we get information about number of pages and current page. We have:
- `Paginate` (this one actually has also `change_page`),
- `Swipable` provides `get_internal_page_count`,  
- `InternallySwipable` provides `current_page` and `num_pages`
- then we have `PageMap`
- I needed to add a similar function to `Paragraphs` recently
- probably more..